code/framework/api/src/main/java/io/cattle/platform/api/resource/AbstractObjectResourceManager.java

```diff
@@ -458,7 +458,7 @@ protected boolean isValidAction(Object obj, Action action) {
             return false;
         }
 
-        if (!StringUtils.isBlank(state) && !state.equals(io.cattle.platform.object.util.ObjectUtils.getState(obj))) {
+        if (!StringUtils.isBlank(state) && !state.contains(io.cattle.platform.object.util.ObjectUtils.getState(obj))) {
```

Comment:
This is a bit hacky.  State is an array, but you're doing toString() on an array and then `contains()`.  The problem here is that other types have `state` as a string.  So depending on the type `state` is a string or array.  How about you add a new field called `states` that is always an array.